`wasm-bindgen`’s `std` feature is non-additive, breaking some build configurations

### Steps to Reproduce

```bash
cargo init --lib
cargo add wasm-bindgen@0.2.96
cargo add wasm-bindgen-futures@0.4.46 --no-default-features
cargo check --target=wasm32-unknown-unknown
```

### Expected Behavior

The project should build

### Actual Behavior

```
error[E0432]: unresolved import `wasm_bindgen::__rt::LazyCell`
    --> /Users/kpreid/.cargo/registry/src/index.crates.io-6f17d22bba15001f/js-sys-0.3.73/src/lib.rs:6041:13
     |
6041 |         use wasm_bindgen::__rt::LazyCell;
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `LazyCell` in `__rt`
     |
note: found an item that was configured out
    --> /Users/kpreid/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wasm-bindgen-0.2.96/src/lib.rs:1598:16
     |
1598 |     pub struct LazyCell<T, F = fn() -> T>(::once_cell::unsync::Lazy<T, F>);
     |                ^^^^^^^^
note: the item is gated here
    --> /Users/kpreid/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wasm-bindgen-0.2.96/src/lib.rs:1597:5
     |
1597 |     #[cfg(not(feature = "std"))]
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

### Additional Context

This problem is occurring because `wasm-bindgen` is exporting the item `LazyCell` only if the feature `wasm-bindgen/std` feature is not enabled, but `wasm-bindgen-futures` uses `LazyCell` if `wasm-bindgen-futures/std` feature is not enabled. But it is not true that `wasm-bindgen/std` implies `wasm-bindgen-futures/std`, so this can fail to compile if `wasm-bindgen/std` is enabled and `wasm-bindgen-futures/std` is not.

Like all Cargo packages, `wasm-bindgen` must ensure that its `std` feature is [additive](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification) — only adds items, without removing any — or if necessary, use some other technique like exporting a macro by which `wasm-bindgen-futures` can make its code conditional on whether `wasm-bindgen/std` is enabled rather than whether `wasm-bindgen-futures/std` is enabled.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,15 @@
 # `wasm-bindgen` Change Log
 --------------------------------------------------------------------------------
 
+## Unreleased
+
+### Fixed
+
+* Fixed `js-sys` and `wasm-bindgen-futures` relying on internal paths of `wasm-bindgen` that are not crate feature additive.
+  [#4305](https://github.com/rustwasm/wasm-bindgen/pull/4305)
+
+--------------------------------------------------------------------------------
+
 ## [0.2.96](https://github.com/rustwasm/wasm-bindgen/compare/0.2.95...0.2.96)
 
 Released 2024-11-29

--- a/crates/futures/src/queue.rs
+++ b/crates/futures/src/queue.rs
@@ -118,11 +118,19 @@ impl Queue {
 
     #[cfg(not(feature = "std"))]
     pub(crate) fn with<R>(f: impl FnOnce(&Self) -> R) -> R {
-        use wasm_bindgen::__rt::LazyCell;
+        use once_cell::unsync::Lazy;
+
+        struct Wrapper<T>(Lazy<T>);
+
+        #[cfg(not(target_feature = "atomics"))]
+        unsafe impl<T> Sync for Wrapper<T> {}
+
+        #[cfg(not(target_feature = "atomics"))]
+        unsafe impl<T> Send for Wrapper<T> {}
 
         #[cfg_attr(target_feature = "atomics", thread_local)]
-        static QUEUE: LazyCell<Queue> = LazyCell::new(Queue::new);
+        static QUEUE: Wrapper<Queue> = Wrapper(Lazy::new(Queue::new));
 
-        f(&QUEUE)
+        f(&QUEUE.0)
     }
 }

--- a/crates/js-sys/Cargo.toml
+++ b/crates/js-sys/Cargo.toml
@@ -25,6 +25,7 @@ default = ["std"]
 std = ["wasm-bindgen/std"]
 
 [dependencies]
+once_cell = { version = "1.12", default-features = false }
 wasm-bindgen = { path = "../..", version = "=0.2.96", default-features = false }
 
 [target.'cfg(target_arch = "wasm32")'.dev-dependencies]

--- a/crates/js-sys/src/lib.rs
+++ b/crates/js-sys/src/lib.rs
@@ -6038,12 +6038,20 @@ pub fn global() -> Object {
     }
     #[cfg(not(feature = "std"))]
     {
-        use wasm_bindgen::__rt::LazyCell;
+        use once_cell::unsync::Lazy;
+
+        struct Wrapper<T>(Lazy<T>);
+
+        #[cfg(not(target_feature = "atomics"))]
+        unsafe impl<T> Sync for Wrapper<T> {}
+
+        #[cfg(not(target_feature = "atomics"))]
+        unsafe impl<T> Send for Wrapper<T> {}
 
         #[cfg_attr(target_feature = "atomics", thread_local)]
-        static GLOBAL: LazyCell<Object> = LazyCell::new(get_global_object);
+        static GLOBAL: Wrapper<Object> = Wrapper(Lazy::new(get_global_object));
 
-        return GLOBAL.clone();
+        return GLOBAL.0.clone();
     }
 
     fn get_global_object() -> Object {

--- a/crates/test/Cargo.toml
+++ b/crates/test/Cargo.toml
@@ -11,7 +11,7 @@ version = "0.3.46"
 
 [features]
 default = ["std"]
-std = ["wasm-bindgen/std", "js-sys/std", "wasm-bindgen-futures/std", "once_cell/std", "scoped-tls"]
+std = ["wasm-bindgen/std", "js-sys/std", "wasm-bindgen-futures/std", "scoped-tls"]
 
 [dependencies]
 gg-alloc = { version = "1.0", optional = true }

